src/new/sendTransaction.js

```diff
@@ -168,9 +169,69 @@ class SendTransaction extends EventEmitter {
       fullTxData.outputs = [...fullTxData.outputs, ...ret.data.outputs];
     }
 
+    this.fullTxData = fullTxData;
+    return fullTxData;
+  }
+
+  /**
+   * Prepare transaction data from inputs and outputs
+   * Fill the inputs if needed, create output change if needed and sign inputs
+   *
+   * @throws SendTxError
+   *
+   * @return {Transaction} Transaction object prepared to be mined
+   *
+   * @memberof SendTransaction
+   * @inner
+   */
+  prepareTx() {
+    if (!this.fullTxData) {
+      this.prepareTxData();
+    }
+    let preparedData = null;
+    try {
+      preparedData = transaction.prepareData(this.fullTxData, this.pin);
+      this.transaction = helpers.createTxFromData(preparedData, this.network);
+      return this.transaction;
+    } catch(e) {
+      const message = oldHelpers.handlePrepareDataError(e);
+      throw new SendTxError(message);
+    }
+  }
+
+  /**
+   * Prepare transaction to be mined from signatures
+   *
+   * The full tx data should already be prepared
+   * since the signatures have already been made
+   *
+   * @throws SendTxError
+   *
+   * @return {Transaction} Transaction object prepared to be mined
+   *
+   * @memberof SendTransaction
+   * @inner
+   */
+  prepareTxFrom(signatures) {
+    if (this.fullTxData === null) {
+      // This method can only be called with a prepared tx data
+      // because prepareTxData may modify the inputs and outputs
+      throw new SendTxError(ErrorMessages.TRANSACTION_IS_NULL);
+    }
+
+    // add each input data from signature
+    const keys = wallet.getWalletData().keys;
+    for (const [index, input] of this.fullTxData.inputs.entries()) {
+      const signature = Buffer.from(signatures[index]);
```

Comment:
I'm a bit confused about this `signatures` parameter. Is is a string or an array of bytes, or an array of string? We should add the type in the docstring with an example and explain that this method is mainly used to sign transactions with signatures that come from ledger.

Reply:
Changed the doctring to reflect what should be the parameter (an array of Buffer, each being a signature). This `Buffer.from` ends up being redundant so I removed it.